lib/clientv3/ippool_e2e_test.go

```diff
@@ -705,6 +705,98 @@ var _ = testutils.E2eDatastoreDescribe("IPPool tests", testutils.DatastoreAll, f
 			Expect(err.Error()).To(ContainSubstring("IPPool(ippool4) CIDR overlaps with IPPool(ippool1) CIDR 1.2.3.0/24"))
 		})
 	})
+
+	Describe("Verify pool blocksize validation", func() {
+		var err error
+		var c clientv3.Interface
+
+		BeforeEach(func() {
+			c, err = clientv3.New(config)
+			Expect(err).NotTo(HaveOccurred())
+
+			be, err := backend.NewClient(config)
+			Expect(err).NotTo(HaveOccurred())
+			be.Clean()
+		})
+
+		It("should prevent the blocksize being changed on an update", func() {
+			By("Creating a pool")
+			pool, err := c.IPPools().Create(ctx, &apiv3.IPPool{
+				ObjectMeta: metav1.ObjectMeta{Name: "ippool1"},
+				Spec: apiv3.IPPoolSpec{
+					CIDR:      "1.2.3.0/24",
+					BlockSize: 25,
+				},
+			}, options.SetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Attempting to change the blockSize")
+			pool.Spec.BlockSize = 26
+			_, err = c.IPPools().Update(ctx, pool, options.SetOptions{})
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(BeAssignableToTypeOf(errors.ErrorValidation{}))
+			Expect(err.Error()).To(ContainSubstring("IPPool BlockSize cannot be modified"))
+		})
+
+		It("should prevent pools from being created with bad block sizes", func() {
+			_, err := c.IPPools().Create(ctx, &apiv3.IPPool{
+				ObjectMeta: metav1.ObjectMeta{Name: "ippool1"},
+				Spec: apiv3.IPPoolSpec{
+					CIDR:      "1.2.3.0/24",
+					BlockSize: 19,
+				},
+			}, options.SetOptions{})
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(BeAssignableToTypeOf(errors.ErrorValidation{}))
+			Expect(err.Error()).To(ContainSubstring("block size must be between"))
+
+			_, err = c.IPPools().Create(ctx, &apiv3.IPPool{
+				ObjectMeta: metav1.ObjectMeta{Name: "ippool1"},
+				Spec: apiv3.IPPoolSpec{
+					CIDR:      "1.2.3.0/24",
+					BlockSize: 33,
+				},
+			}, options.SetOptions{})
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(BeAssignableToTypeOf(errors.ErrorValidation{}))
+			Expect(err.Error()).To(ContainSubstring("block size must be between"))
+		})
+
+		It("should prevent the creation of a pool with an identical or overlapping CIDR using block sizes", func() {
```

Comment:
Do we have a test that checks the mismatched block size logic?

e.g.
- Create a pool, blocksize X
- Allocate an IP (and thus a block) from that pool
- Delete the first pool.
- Recreate the pool with blocksize Y, see that it fails.
- Release the IP allocation
- Recreate the pool with blocksize Y, see that it succeeds.

Reply:
I thought I already covered that - https://github.com/projectcalico/libcalico-go/pull/935/files#diff-0912dca51c2f14012b137b66c0b4babdR727

Reply:
Yep, looks like it - I just couldn't remember if it was already there or not.